test/engine/enginerpc/enginerpc.cc

```diff
@@ -150,6 +153,182 @@ TEST_F(EngineRpc, GetVersion) {
   erpc.shutdown();
 }
 
+TEST_F(EngineRpc, GetNbrHost) {
+  std::ostringstream oss;
+  oss << "1";
+  enginerpc erpc("0.0.0.0", 40001);
+  std::atomic<bool> continuerunning(true);
+
+  auto fn = [&continuerunning] () {
+	while(continuerunning) {
+	  command_manager::instance().execute();
+	  std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    }
+  };
+
+  std::thread th(fn);
+  auto output = execute("GetNbrHost");
+  continuerunning = false;
+  th.join();
```

Comment:
Don't you want to see the thread running while it has not its result?

I have the feeling that the main thread will stop the thread when needed...
Synchronization with tempo is not a good thing